### PR TITLE
Project setup

### PR DESCRIPTION
**Pull request by Alexander-Eryomenko**
project setup

### CODE REVIEW

File: src/components/ComicsPage/ComicsPage.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  comics: Array<Comic>
+  isLoading: boolean
+}
+
+interface IComicPageItemProps {
+  description: string
+  thumbnail: IThumbnail
+}
+
+class ComicsPage extends React.Component<any, IState> {
```
Comment (alinalaktionova) [inline src/components/ComicsPage/ComicsPage.tsx]: remove any

File: src/components/HeroesItem/HeroesItem.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+}
+export const HeroesItem = (props: IProps) => {
+  const { hero } = props
+
+
+  return (
+    <div className="heroes">
+
+      <div className="heroes__avatar">
+          <img src={`${hero.thumbnail.path}.${hero.thumbnail.extension}`} alt="avatar" />
```
Comment (alinalaktionova) [inline src/components/HeroesItem/HeroesItem.tsx]: better to create separate variable for image src

File: src/components/HeroesItem/heroesItem.scss [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        border-radius: 50%;
+        width: 100%;
+        height: 100%;
+      }
+  }
+  
+  &__info {
+    margin-left: 40px;
+  }
+  &__btn {
```
Comment (alinalaktionova) [inline src/components/HeroesItem/heroesItem.scss:25]: empty line is needed between 2 style blocks

File: src/components/MarvelApi/MarvelApi.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import { Comic } from '../../types/hero'
+
+export class MarvelApi {
+  static url = 'http://gateway.marvel.com/v1/public/'
+  static privateApiKey = process.env.REACT_APP_PRIVATE_API_KEY
+  static publicApiKey = process.env.REACT_APP_PUBLIC_KEY
+  static hash = process.env.REACT_APP_HASH
+
+  static async getCharacters (nameCharacter?: string) {
+    let url = `${MarvelApi.url}characters?ts=1&apikey=${MarvelApi.publicApiKey}&hash=${MarvelApi.hash}&limit=5`
```
Comment (alinalaktionova) [inline src/components/MarvelApi/MarvelApi.tsx]: use axios params for query parameters

File: src/components/MarvelApi/MarvelApi.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    if (nameCharacter) {
+      
+      url += `&nameStartsWith=${nameCharacter}`
+    }
+    const response = await axios.get(url)
+    const data = await response.data.data.results
```
Comment (alinalaktionova) [inline src/components/MarvelApi/MarvelApi.tsx]: data variable is redundant here

File: src/components/MarvelApi/index.ts [hunk 6]
```diff
@@ -0,0 +1 @@
+export {MarvelApi} from './MarvelApi'
```
Comment (alinalaktionova) [inline src/components/MarvelApi/index.ts:1]: config eslint to add spacing between {}

File: src/components/PageError/PageError.tsx [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import './pageError.scss'
+import error from '../../img/error.jpg'
+import { withRouter } from "react-router-dom"
+
+const PageError = (props: any) => {
```
Comment (alinalaktionova) [inline src/components/PageError/PageError.tsx:6]: any

File: src/components/PageError/PageError.tsx [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
+import './pageError.scss'
+import error from '../../img/error.jpg'
+import { withRouter } from "react-router-dom"
```
Comment (alinalaktionova) [inline src/components/PageError/PageError.tsx]: use single quotes or double quotes throughout all code. Choose one option

File: src/components/app/App.tsx [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+        <>
+          <Router>
+            <Header/>
+            <Switch>
+              <Route exact path="/" component={Home} />
+              <Route exact path="/comics/:id" component={ComicsPage} />
+              <Route path="*" component={PageError} />
+            </Switch>
+          </Router>
+</>
```
Comment (alinalaktionova) [inline src/components/app/App.tsx]: code style

File: src/components/Spinner/spiner.scss [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+      right: 6px;
+      bottom: 6px;  
+      animation: spin 4s linear infinite;
+    }
+  }
+}
+
+@keyframes spin{
+  0% {transform: rotate(0deg);}
+  100% {transform: rotate(360deg);}
```
Comment (alinalaktionova) [inline src/components/Spinner/spiner.scss]: spacing

File: src/components/heroes/Heroes.tsx [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+
+import './heroes.scss'
+
+interface IState {
+  heroes: Array<Hero>
+  isLoading: boolean
+  search: string
+}
+
+class Heroes extends React.Component<any, IState> {
```
Comment (alinalaktionova) [inline src/components/heroes/Heroes.tsx]: any

File: src/components/heroes/Heroes.tsx [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+  handleSearch = (event: React.ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    this.props.history.push(`?query=${this.state.search}`)
+  }
+
+  render() {
+    const { isLoading } = this.state
+    return (
+      <div className="container">
+        <SearchPanel onSubmit={this.handleSearch} onChange={this.onInput} value={this.state.search} />
```
Comment (alinalaktionova) [inline src/components/heroes/Heroes.tsx]: no need to pass these props to the component, better to handle them right in SearchPanel component

File: src/types/hero.ts [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+  extension: string
+}
+export type Hero = {
+  id: number
+  description: string
+  name: string
+  thumbnail: IThumbnail
+}
+
+export type Comic = {
```
Comment (alinalaktionova) [inline src/types/hero.ts]: Hero and Comics types are the same, you can use interface inheritance to remove duplication

**Review by alinalaktionova — CHANGES_REQUESTED** [review]

File: src/components/HeroesItem/heroesItem.scss (hunk 3 above)
Comment (Alexander-Eryomenko) [reply src/components/HeroesItem/heroesItem.scss:25]: done

File: src/components/ComicsPage/ComicsPage.tsx (hunk 1 above)
Comment (Alexander-Eryomenko) [reply src/components/ComicsPage/ComicsPage.tsx]: done

File: src/components/HeroesItem/HeroesItem.tsx (hunk 2 above)
Comment (Alexander-Eryomenko) [reply src/components/HeroesItem/HeroesItem.tsx]: done

File: src/components/MarvelApi/MarvelApi.tsx (hunk 5 above)
Comment (Alexander-Eryomenko) [reply src/components/MarvelApi/MarvelApi.tsx]: done

File: src/components/app/App.tsx (hunk 9 above)
Comment (Alexander-Eryomenko) [reply src/components/app/App.tsx]: done

File: src/components/Spinner/spiner.scss (hunk 10 above)
Comment (Alexander-Eryomenko) [reply src/components/Spinner/spiner.scss]: done

File: src/components/heroes/Heroes.tsx (hunk 11 above)
Comment (Alexander-Eryomenko) [reply src/components/heroes/Heroes.tsx]: done

File: src/components/MarvelApi/MarvelApi.tsx (hunk 4 above)
Comment (Alexander-Eryomenko) [reply src/components/MarvelApi/MarvelApi.tsx]: done

File: src/types/hero.ts (hunk 13 above)
Comment (Alexander-Eryomenko) [reply src/types/hero.ts]: done

**Pull request merged**